### ISSUE TITLE
Convert Dates into Numeric Week Counts

### ISSUE THREAD

**Issue by brichard1638** [oobianom/quickcode#24]
While an R package called lubridate contains a function called **week** that allows the user to convert dates into numeric week counts, this functionality can and should be expanded to include two additional calculations that return various week count sequencing. It is therefore proposed that a new function called **getWeekSeq** be added to the quickcode package.

The nuances are largely contained within how the numeric week values are sequenced between years.

For example, the function below calculates the standard sequence of numeric values for weeks from a set of passed dates:
`getWeekSeq <- function(start_date, end_date) {`
  `# Convert input dates to Date objects`
  `start_date <- as.Date(start_date)`
  `end_date <- as.Date(end_date)`
  
  `# Generate a sequence of dates`
  `date_sequence <- seq(from = start_date, to = end_date, by = "day")`
  
  `# Create a data frame`
  `week_df <- data.frame(DATE = date_sequence)`
  
  `# Add a column for the week's sequence starting with 1`
  `sequence = lubridate::week(week_df$DATE)`
  `week_df$WEEK <- paste("WEEK", sequence)`
  
  `return(week_df)`
`}`

This numeric week sequence is referred to as **by week** and processes the week count between years based on the previous year's last weekday (12/31). For example, if 12/31 of 2024 ends on a Tuesday, then Wednesday starts the new week beginning with a numeric count of 1. A sequential example is provided below:

2025-12-29  WEEK 52
2025-12-30  WEEK 52
2025-12-31  WEEK 53
2026-01-01  WEEK 1
2026-01-02  WEEK 1
2026-01-03  WEEK 1
2026-01-04  WEEK 1
2026-01-05  WEEK 1
2026-01-06  WEEK 1
2026-01-07  WEEK 1
2026-01-08  WEEK 2

A second sequence option is to number each week within a date range in the same manner as described above, except that each week is sequenced on exactly each Monday, irrespective of what day at the end of the year the 31st falls. This sequential pattern is referred to as **by sequence.** A sequential example of this pattern is provided below:

2025-12-29  WEEK 52
2025-12-30  WEEK 52
2025-12-31  WEEK 52
2026-01-01  WEEK 52
2026-01-02  WEEK 52
2026-01-03  WEEK 52
2026-01-04  WEEK 52
2026-01-05  WEEK 1
2026-01-06  WEEK 1
2026-01-07  WEEK 1
2026-01-08  WEEK 1

A third and final sequential option provides a continuous weekly sequence between years irrespective of how many years are captured between a start and and an end date. Like the **by sequence** method, this process, too starts each weekly sequence on a Monday. This sequential pattern is referred to as **by continuous.** A sequential example of this pattern is provided below:

2025-12-29  WEEK 52
2025-12-30  WEEK 52
2025-12-31  WEEK 52
2026-01-01  WEEK 52
2026-01-02  WEEK 52
2026-01-03  WEEK 52
2026-01-04  WEEK 52
2026-01-05  WEEK 53
2026-01-06  WEEK 53
2026-01-07  WEEK 53
2026-01-08  WEEK 53

It is proposed that a function titled, **getWeekSeq** be created that contains a total of (3) arguments:
getWeekSeq(start_date, end_date, seq)

start_date = A scaler of class Date
end_date = A scaler of class Date; must be later than the start_date
seq = The sequential method used to generate the data frame; the options for configuring this argument are as follows:
- bywk
- byseq
- bycont

**Comment by oobianom** [comment]
Working on this now. I have committed initially the function you have up there for now.

**Comment by oobianom** [comment]
Haven't done much this month to have a release. My plan for us to have the next release  between mid and end of June.

**Comment by brichard1638** [comment]
Okay. I'm not sure what your availability is to consider any new functions for the upcoming quickcode release.

I'm looking forward to the next quickcode release, especially the function supporting numeric week counts.
________________________________

**Comment by oobianom** [comment]
Hey Brice, is this a good new function or it already exist - check https://github.com/oobianom/quickcode/issues/26

**Comment by brichard1638** [comment]
I noticed that you did not rewrite this function to include its most critical argument: the seq argument. All I provided was a skeleton view of the function, not the entire function. I was hoping you would rewrite the function to include the seq argument. The function in its current form in 0.9 is worthless and has no value proposition because it lacks this seq argument. I am strongly recommending that this function be removed from the 0.9 version of quickcode.

**Comment by oobianom** [comment]
Yes, I forgot that inclusion. No worries, we will develop it further for v1.0

**Comment by brichard1638** [comment]
Are you going to re-publish quickcode so that the version is baselined? As you know some of the functions have been removed which remain in the 0.9 version of the package.

You should probably resubmit the package again just to baseline it. This action will also correctly align the .pdf manual with currently available functions.
________________________________

**Comment by oobianom** [comment]
Re-publishing it now will up the version. which functions are you referring to that still remain in version 0.9? Ideally, I want to do any needed updates in the github repository, then we all is set, we can publish 1.0

**Comment by oobianom** [comment]
Brice, going back to this function, I am still not sure how this "seq" argument can be programmed. Do you have a code for us to implement the "seq"
